book_source/03_topical_pages/07_remote_access/01_pecan_api.Rmd

```diff
@@ -541,6 +553,220 @@ print(json.dumps(response.json(), indent=2))
+##   "format_id": 19,
+##   "name": "AmeriFlux.level4.h",
+##   "notes": "Half-hourly AmeriFlux level 4 gap filled, partitioned, and flagged flux tower data. Variables description:  ## Level 4 data are obtained from the level 3 products, data are ustar filtered, gap-filled using different methods (ANN and ## MDS) and partitioned (i.e. NEE, GPP, and Re). Flags with information regarding quality of the original and gapfilled data ## are added.  Missing values: -9999.",
+##   "header": "",
+##   "format_variables": [
+##     {
+##       "description": "Latent heat flux",
+##       "name": "LE_f",
+##       "unit": "W m-2"
+##     },
+##     ...
+##   ]
+## }
+```
+
+### {-}
+
+### `GET /api/inputs/` {.tabset .tabset-pills}
+
+#### R Snippet
+
+```R
+# Get the inputs needed for a workflow with model_id = 1000000022 & site_id = 676
+res <- httr::GET(
+         "http://localhost:8000/api/inputs/?model_id=1000000022&site_id=676",
+         httr::authenticate("carya", "illinois")
+       )
+print(jsonlite::fromJSON(rawToChar(res$content)))
+```
+```
+## $inputs
+##                sitename model_name revision tag                      hostname file_name   format_name mimetype
+## 1 Willow Creek (US-WCr)     SIPNET      ssr met ebi-forecast.igb.illinois.edu  wcr.clim Sipnet.climna text/csv
+## 2 Willow Creek (US-WCr)     SIPNET      ssr met                        docker  wcr.clim Sipnet.climna text/csv
+##                   file_path         id input_name          start_date            end_date
+## 1 /home/share/data/dbfiles/        235            2000-01-01 06:00:00 2007-01-01 05:59:00
+## 2        /data/sites/willow 2000000001            1998-01-01 00:00:00 2006-12-31 00:00:00
+
+## $count
+## [1] 2
+```
+
+#### Python Snippet
+
+```python
+# Get the inputs needed for a workflow with model_id = 1000000022 & site_id = 676
+response = requests.get(
+             "http://localhost:8000/api/inputs/?model_id=1000000022&site_id=676",
+             auth=HTTPBasicAuth('carya', 'illinois')
+           )
+print(json.dumps(response.json(), indent=2))
+```
+```
+## {
+##   "inputs": [
+##     {
+##       "sitename": "Willow Creek (US-WCr)",
+##       "model_name": "SIPNET",
+##       "revision": "ssr",
+##       "tag": "met",
+##       "hostname": "ebi-forecast.igb.illinois.edu",
+##       "file_name": "wcr.clim",
+##       "format_name": "Sipnet.climna",
+##       "mimetype": "text/csv",
+##       "file_path": "/home/share/data/dbfiles/",
+##       "id": 235,
+##       "input_name": "",
+##       "start_date": "2000-01-01 06:00:00",
+##       "end_date": "2007-01-01 05:59:00"
+##     },
+##     ...
+##   ],
+##   "count: 2
+## }
+```
+
+### {-}
+
+### `GET /api/inputs/{input_id}` {.tabset .tabset-pills}
+
+#### R Snippet
+
+```R
+# Download the input file with id = 99000000003
+res <- httr::GET(
+         "http://localhost:8000/api/inputs/99000000003",
+         httr::authenticate("carya", "illinois")
+       )
+writeBin(res$content, "test.2002.nc")
```

Comment:
When we use the Input query like this, how are we supposed to know what the filename should be? If not provided in an example here, you should definitely make sure there are helper functions in the pecan package that does both the GET and the query for the filenames and the write. @robkooper beyond this PR, but I think it would be great if there were then core PEcAn functions that would not just use the API to grab files, but then stick them into the local DB as new dbfiles associated with the correct Input record. Would really like to get to the point where PEcAn nodes default to grabbing files from each other before attempting to download and reprocess the raw Inputs from their original sources (e.g. I could grab the met I need for a run from Shawn or Ankur if they've already run that site)